### PR TITLE
Add new models to Rafiki & add license to md file

### PR DESCRIPTION
**Pull request by pinpom**

### CODE REVIEW

File: examples/models/image_classification/LeNet5.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+        tensorboard = TensorBoard(log_dir="logs/{}".format(time()))
+        self._model.fit_generator(train_generator, steps_per_epoch=steps_per_epoch, epochs=ep, 
+                    validation_data=validation_generator, validation_steps=validation_steps, 
+                    shuffle=True, callbacks=[tensorboard])
+
+        # Compute train accuracy
+        (train_loss, train_acc) = self._model.evaluate(X_validation, y_validation)
+
+    def evaluate (self, dataset_uri):
```
Comment (nginyc) [inline examples/models/image_classification/LeNet5.py]: Should not have space after method names

File: examples/models/image_classification/LeNet5.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+    def load_parameters(self, params):
+        # Load model parameters
+        model_base64 = params.get('model_base64', None)
+        if model_base64 is None:
+            raise InvalidModelParamsException()
+        
+        model_bytes = base64.b64decode(params['model_base64'].encode('utf-8'))
+        self._model = pickle.loads(model_bytes)
+    def _prepare_X(self, images):
```
Comment (nginyc) [inline examples/models/image_classification/LeNet5.py:143]: Should add some spacing between all methods

File: README.md [hunk 3]
```diff
@@ -1,3 +1,21 @@
+<!--
```
Comment (nginyc) [inline README.md]: Is it appropriate/required to add license in the README here? Doesn't show up in README in its default view. And we have already have a `LICENSE` which Github detects automatically (https://github.com/nginyc/rafiki/blob/master/LICENSE)

File: examples/models/image_classification/CNN.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+class CNN(BaseModel):
+    '''
+    Implements CNN using keras for simple image classification
+    '''
+    @staticmethod
+    def get_knob_config():
+        return {
+            'epochs': FixedKnob(100),
+            'batch_size': FixedKnob(86),
```
Comment (nginyc) [inline examples/models/image_classification/CNN.py]: Given that one of the strengths of Rafiki automating the hyperparameter tuning process for models, and fine-tuning it for application developers' datasets, I think we should add more knobs here and set appropriate value ranges e.g. `batch_size` can be `IntegerKnob`.

File: examples/models/image_classification/LeNet5.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+class LeNet5(BaseModel):
+    '''
+    Implements LeNet5 network to train model for simple image classification
+    '''
+    @staticmethod
+    def get_knob_config():
+        return {
+            'epochs': FixedKnob(10),
+            'batch_size': FixedKnob(128)
```
Comment (nginyc) [inline examples/models/image_classification/LeNet5.py]: Same comment here as for the `get_knob_config` of the CNN model

File: examples/models/image_classification/LeNet5.py [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    test_model_class(
+        model_file_path=__file__,
+        model_class='LeNet5',
+        task=TaskType.IMAGE_CLASSIFICATION,
+        dependencies={
+            ModelDependency.TENSORFLOW: '1.12.0',
+            ModelDependency.KERAS: '2.2.4'
+        },
+    train_dataset_uri='data/fashion_mnist_for_image_classification_train.zip',
+    test_dataset_uri='data/fashion_mnist_for_image_classification_test.zip',
```
Comment (nginyc) [inline examples/models/image_classification/LeNet5.py]: These parameters no longer works on `dev`. Need to update them. Refer to sample models on `dev` e.g. https://github.com/nginyc/rafiki/blob/dev/examples/models/image_classification/SkDt.py

File: README.md (hunk 3 above)
Comment (pinpom) [reply README.md]: i saw a comment by @nudles at https://github.com/nginyc/rafiki/pull/96 so I guess it is supposed to be added in README.md file

**Comment by pinpom** [comment]
@nginyc: editted both models according to your comments. tested again on dev branch. successfully ran at acc ~ 90%

File: README.md (hunk 3 above)
Comment (nudles) [reply README.md]: If there is no side effect (e.g., the document generator cannot parse the comments), then just add the license in every file created by us.

File: examples/models/image_classification/CNN.py [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+from keras.preprocessing.image import ImageDataGenerator
+from keras.callbacks import ReduceLROnPlateau
+
+from rafiki.model import BaseModel, InvalidModelParamsException, test_model_class, \
+                        CategoricalKnob, FixedKnob, dataset_utils, logger
+from rafiki.constants import TaskType, ModelDependency
+
+class CNN(BaseModel):
+    '''
+    Implements CNN using keras for simple image classification
```
Comment (nginyc) [inline examples/models/image_classification/CNN.py]: Specifically which CNN is this? I believe we should rename this model to its exact CNN name. E.g. LeNet5, VGG are specific CNNs with a well-defined architecture

File: examples/models/image_classification/CNN.py [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+        model.add(Dropout(0.25))
+        model.add(Conv2D(filters = 64, kernel_size = (3,3),padding = 'Same', activation ='relu'))
+        model.add(Conv2D(filters = 64, kernel_size = (3,3),padding = 'Same', activation ='relu'))
+        model.add(layers.MaxPooling2D((2, 2)))
+        model.add(Dropout(0.25))
+        model.add(Flatten())
+        model.add(Dense(256, activation = "relu"))
+        model.add(BatchNormalization())
+        model.add(Dense(10, activation = "softmax"))
+        my_optimizer = RMSprop(lr=0.001, rho=0.9, epsilon=1e-08, decay=0.0)
```
Comment (nginyc) [inline examples/models/image_classification/CNN.py]: For both models, can we add learning rate `lr` to the knob configuration? It is probably one of the most important hyperparameters (ref https://machinelearningmastery.com/understand-the-dynamics-of-learning-rate-on-deep-learning-neural-networks/)

File: examples/models/image_classification/LeNet5.py [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+    test_model_class(
+        model_file_path=__file__,
+        model_class='LeNet5',
+        task=TaskType.IMAGE_CLASSIFICATION,
+        dependencies={
+            ModelDependency.TENSORFLOW: '1.12.0',
+            ModelDependency.KERAS: '2.2.4'
+        },
+    train_dataset_path='data/fashion_mnist_for_image_classification_train.zip',
+    val_dataset_path='data/fashion_mnist_for_image_classification_val.zip',
```
Comment (nginyc) [inline examples/models/image_classification/LeNet5.py]: Try to indent consistently

File: examples/models/image_classification/CNN.py (hunk 7 above)
Comment (pinpom) [reply examples/models/image_classification/CNN.py]: this model's is not any specific well-defined architecture. Its architecture is of basic CNN, but layers are different from other well-defined ones. Believe that model developers can upload their own models on rafiki, so any kinds of models would be welcome as long as the result is acceptable?

Comment (nginyc) [reply examples/models/image_classification/CNN.py]: @nudles what do you think? this is for adding an example model to Rafiki's codebase

File: examples/models/image_classification/CNN.py (hunk 8 above)
Comment (pinpom) [reply examples/models/image_classification/CNN.py]: changed the code accordingly. set `lr` as the knob configuration to be tuned for both models. tested based on new documentation and get ~90% accuracy

**Comment by pinpom** [comment]
git re-base with latest version & changed code of models

File: examples/models/image_classification/CNN.py (hunk 7 above)
Comment (pinpom) [reply examples/models/image_classification/CNN.py]: i just changed to new model
